glide-core/redis-rs/redis/src/cluster_async/connections_container.rs

```diff
@@ -134,11 +140,163 @@ impl<Connection> std::fmt::Display for ConnectionsMap<Connection> {
+            debug!(
+                "RefreshTaskStatus: Notifying tasks before transitioning to ReconnectingTooLong."
+            );
+            notifier.notify();
+            *self = RefreshTaskStatus::ReconnectingTooLong;
+        } else {
+            debug!("RefreshTaskStatus: Already in ReconnectingTooLong status.");
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn notify_waiting_requests(&mut self) {
+        if let RefreshTaskStatus::Reconnecting(notifier) = self {
+            debug!("RefreshTaskStatus::notify_waiting_requests notify");
+            notifier.notify();
+        } else {
+            debug!("RefreshTaskStatus::notify_waiting_requests - ReconnectingTooLong status.");
+        }
+    }
+}
+
+// Struct combining the task handle and its status
+#[derive(Debug)]
+pub(crate) struct RefreshTaskState {
+    pub handle: JoinHandle<()>,
+    pub status: RefreshTaskStatus,
+}
+
+impl RefreshTaskState {
+    // Creates a new `RefreshTaskState` with a `Reconnecting` state and a new notifier.
+    pub fn new(handle: JoinHandle<()>) -> Self {
+        debug!("RefreshTaskState: Creating a new instance with a Reconnecting state.");
+        RefreshTaskState {
+            handle,
+            status: RefreshTaskStatus::new(),
+        }
+    }
+}
+
+impl Drop for RefreshTaskState {
+    fn drop(&mut self) {
+        if let RefreshTaskStatus::Reconnecting(ref notifier) = self.status {
+            debug!("RefreshTaskState: Dropped while in Reconnecting status. Notifying tasks.");
+            notifier.notify();
+        } else {
+            debug!("RefreshTaskState: Dropped while in ReconnectingTooLong status.");
+        }
+
+        // Abort the task handle if it's not yet finished
+        if !self.handle.is_finished() {
+            debug!("RefreshTaskState: Aborting unfinished task.");
+            self.handle.abort();
+        } else {
+            debug!("RefreshTaskState: Task already finished, no abort necessary.");
+        }
+    }
+}
+
+// This struct is used to track the status of each address refresh state
+// TODO move this struct logic into the connection_map itself
+#[derive(Default)]
+pub(crate) struct RefreshConnectionStates {
+    // Follow the refresh ops on the connections
+    pub(crate) refresh_address_in_progress: HashMap<String, RefreshTaskState>,
+}
+
+impl RefreshConnectionStates {
+    // Clears all ongoing refresh connection tasks and resets associated state tracking.
+    //
+    // - This method removes all entries in the `refresh_address_in_progress` map.
+    // - The `Drop` trait is responsible for notifying the associated notifiers and aborting any unfinished refresh tasks.
+    pub(crate) fn clear_refresh_state(&mut self) {
+        debug!(
+            "clear_refresh_state: removing all in-progress refresh connection tasks for addresses: {:?}",
+            self.refresh_address_in_progress.keys().collect::<Vec<_>>()
+        );
+
+        // Clear the entire map; Drop handles the cleanup
```

Comment:
Drop handles the cleanup?

Reply:
Yes, the Drop (i.e. dtor) terminates the tokio task and triggers the notifier.

Reply:
oh so it wasnt clear. you can change to something like "The 'Drop' trait will handle the cleanup of..."